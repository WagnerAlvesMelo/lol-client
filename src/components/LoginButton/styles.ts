import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    border-radius: ${theme.border.radius};
    border: 1px solid ${theme.colors.secondary};
    background-color: transparent;
    width: 6.4rem;
    height: 6.4rem;
    outline: none;
    padding: 1rem;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.red};
      border-color: ${theme.colors.red};

      svg {
        fill: ${theme.colors.mainBg};
      }
    }

    svg {
      fill: ${theme.colors.secondary};
    }
  `}
`
