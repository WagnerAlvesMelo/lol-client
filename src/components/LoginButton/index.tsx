import * as S from './styles'

const LoginButton = ({ ...rest }) => (
  <S.Wrapper {...rest}>
    <svg
      version="1.1"
      id="Capa_1"
      xmlns="http://www.w3.org/2000/svg"
      x="0px"
      y="0px"
      viewBox="0 0 512 512"
    >
      <g>
        <g>
          <polygon
            points="291.838,35.838 263.57,64.106 435.474,236.011 0,236.011 0,275.989 435.474,275.989 263.57,447.894
			291.838,476.162 512,256 		"
          />
        </g>
      </g>
    </svg>
  </S.Wrapper>
)

export default LoginButton
