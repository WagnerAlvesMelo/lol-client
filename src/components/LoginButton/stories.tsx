import { Story, Meta } from '@storybook/react/types-6-0'
import LoginButton from '.'

export default {
  title: 'LoginButton',
  component: LoginButton
} as Meta

export const Default: Story = () => <LoginButton />
