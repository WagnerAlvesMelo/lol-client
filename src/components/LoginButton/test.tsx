import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helper'

import LoginButton from '.'

describe('<LoginButton />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<LoginButton />)

    expect(screen.getByRole('button')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('verify if button action is triggered', async () => {
    const onClick = jest.fn()
    renderWithTheme(<LoginButton onClick={onClick} />)

    expect(onClick).not.toHaveBeenCalled()

    userEvent.click(screen.getByRole('button'))
    await waitFor(() => expect(onClick).toHaveBeenCalledTimes(1))
  })
})
