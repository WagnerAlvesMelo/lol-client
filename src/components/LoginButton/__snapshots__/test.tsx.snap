// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should render the heading 1`] = `
.c0 {
  border-radius: 0.4rem;
  border: 1px solid #EDEDED;
  background-color: transparent;
  width: 3.4rem;
  height: 3.4rem;
  outline: none;
  padding: 0.5rem;
  cursor: pointer;
}

.c0:hover {
  background-color: #FF3232;
  border-color: #FF3232;
}

.c0:hover svg {
  fill: white;
}

.c0 svg {
  fill: #EDEDED;
}

<button
  class="c0"
>
  <svg
    id="Capa_1"
    version="1.1"
    viewBox="0 0 512 512"
    x="0px"
    xmlns="http://www.w3.org/2000/svg"
    y="0px"
  >
    <g>
      <g>
        <polygon
          points="291.838,35.838 263.57,64.106 435.474,236.011 0,236.011 0,275.989 435.474,275.989 263.57,447.894 291.838,476.162 512,256 		"
        />
      </g>
    </g>
  </svg>
</button>
`
